core/src/impl/Kokkos_Half_MathematicalFunctions.hpp

```diff
@@ -315,6 +321,100 @@ KOKKOS_INLINE_FUNCTION bool isnan(Kokkos::Experimental::bhalf_t x) {
   return Impl::impl_isnan(x);
 }
 #endif
+
+#if !(defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_COMPILER_MSVC))
+template <typename fp16_t>
+KOKKOS_INLINE_FUNCTION fp16_t nextafter_half_impl(fp16_t from, fp16_t to) {
```

Comment:
Define the helper function in `Impl::`
`Kokkos::nextafter_half_impl -> Kokkos::Impl::nextafter_half_helper` or whatnot